automation/src/test/scala/org/broadinstitute/dsde/test/api/SnapshotAPISpec.scala

```diff
@@ -0,0 +1,143 @@
+              throw ex
+          }
+          assume(drSnapshots.getItems.size() == 2,
+            s"TDR at $dataRepoBaseUrl did not have 2 snapshots for this test to use!")
+
+          val dataRepoSnapshotId = drSnapshots.getItems.get(0).getId
+          val anotherDataRepoSnapshotId = drSnapshots.getItems.get(1).getId
+
+          logger.info(s"found 2 snapshots from $dataRepoBaseUrl as user ${owner.email}: $dataRepoSnapshotId, $anotherDataRepoSnapshotId")
+
+          // add snapshot reference to the workspace. Under the covers, this creates the workspace in WSM and adds the ref
+          createSnapshotReference(projectName, workspaceName, dataRepoSnapshotId, "firstSnapshot")
+
+          // validate the snapshot was added correctly: list snapshots in Rawls, should return 1, which we just added.
+          // if we can successfully list snapshot references, it means WSM created its copy of the workspace
+          val firstListResponse = listSnapshotReferences(projectName, workspaceName)
+          val firstResources = Rawls.parseResponseAs[DataReferenceList](firstListResponse).getResources.asScala
+          firstResources.size shouldBe 1
+          firstResources.head.getName shouldBe "firstSnapshot"
+          firstResources.head.getReferenceType shouldBe ReferenceTypeEnum.DATA_REPO_SNAPSHOT
+          firstResources.head.getReference.getSnapshot shouldBe dataRepoSnapshotId
+
+          // add a second snapshot reference to the workspace. Under the covers, this recognizes the workspace
+          // already exists in WSM, so it just adds the ref
+          createSnapshotReference(projectName, workspaceName, anotherDataRepoSnapshotId, "secondSnapshot")
+
+          // validate the second snapshot was added correctly: list snapshots in Rawls, should return 2, which we just added
+          val secondListResponse = listSnapshotReferences(projectName, workspaceName)
+          // sort by reference name for easy predictability inside this test: "firstSnapshot" is before "secondSnapshot"
+          val secondResources = Rawls.parseResponseAs[DataReferenceList](secondListResponse)
+            .getResources.asScala.sortBy(_.getName)
+          secondResources.size shouldBe 2
+          secondResources.head.getName shouldBe "firstSnapshot"
+          secondResources.head.getReference.getSnapshot shouldBe dataRepoSnapshotId
+          secondResources.head.getReferenceType shouldBe ReferenceTypeEnum.DATA_REPO_SNAPSHOT
+          secondResources(1).getName shouldBe "secondSnapshot"
+          secondResources(1).getReference.getSnapshot shouldBe anotherDataRepoSnapshotId
+          secondResources(1).getReferenceType shouldBe ReferenceTypeEnum.DATA_REPO_SNAPSHOT
+        }
+      }
+    }
+
+  }
+
+
+  private def listSnapshotReferences(projectName: String, workspaceName: String)(implicit authToken: AuthToken) = {
+    Rawls.getRequest(
+      uri = s"${Rawls.url}api/workspaces/$projectName/$workspaceName/snapshots?offset=0&limit=10")
```

Comment:
Could `offset` and `limit` be possibly parameterized for this method...with `0` and `10` being the default values respectively?

Reply:
sure, we could implement that ...  there's no use case for it but I could add it in!

Reply:
Cool. Doesn't need to part of this PR.

Reply:
@ungwudik I added it!